src/d3a/models/area/__init__.py

```diff
@@ -134,7 +134,15 @@ def restore_state(self, saved_state):
         self.stats.restore_state(saved_state)
 
     def area_reconfigure_event(self, **kwargs):
-        if self.strategy is not None:
+        if key_in_dict_and_not_none(kwargs, 'strategy_object'):
+            self.strategy = kwargs['strategy_object']
+            self.activate()
+            # self.strategy.area = self.parent
+            # self.strategy.owner = self
+            self.strategy.event_activate()
+            self.strategy.event_market_cycle()
```

Comment:
I think you can also move all these lines to the event too. I am mentioning this in order to leave the `area_reconfigure_event` completely unaffected from the live events.